kitsune/sumo/static/sumo/js/form-wizard.js

```diff
@@ -69,9 +75,100 @@ export class FormWizard extends HTMLElement {
     this.#updateFormProgress();
   }
 
-  #updateSteps() {
-    this.#steps = [...this.children].map((child) => child.getAttribute("name"));
-    this.#updateFormProgress();
+  /**
+   * @typedef {object} FormWizardStep
+   * @property {string} name
+   *  The name of the step. Used as an identifier to determine which
+   *  child view should be shown and help determine how far the user
+   *  has progressed through the form.
+   * @property {string} status
+   *  Status of the step. One of "done" | "active" | "unavailable".
+   *  Used to hide/show steps and update the step indicator.
+   * @property {string} label
+   *  Label for the form step to be shown in the step indicator.
+   */
+
+  /**
+   * Stores the possible form step names and associated data.
+   *
+   * @param {FormWizardStep[]} nextSteps
+   *  Step data representing the desired state of the wizard.
+   */
+  set steps(nextSteps) {
+    this.#steps = nextSteps;
+    // Determine whether or not we need to update activeStep.
+    let nextActiveStep = nextSteps.find(
+      (step) => step.status === "active"
+    )?.name;
+    if (nextActiveStep) {
+      this.activeStep = nextActiveStep;
+    }
+    // Build or update the steps sidebar.
+    this.#updateStepIndicator();
+  }
+
+  get steps() {
+    return this.#steps;
+  }
+
+  /**
+   * Updates data for the step specified by name and/or changes the currently
+   * active step. Active step only changes when the step getting updated is
+   * currently "unavailable".
+   *
+   * @param {string} name - Name of the step to be updated.
+   * @param {FormWizardStep} data - Data to update the step with.
+   */
+  setStep(name, data) {
+    let currentStatus = this.#steps.find((step) => step.name === name)?.status;
+    let isUnavailable = currentStatus === "unavailable";
+    let nextSteps = this.#steps.map((step) => {
+      if (step.status === "active" && isUnavailable) {
+        return { ...step, status: "done" };
+      } else if (step.name === name && isUnavailable) {
+        return { ...step, ...data, status: "active" };
+      } else if (step.name === name) {
+        return { ...step, ...data };
```

Comment:
I suppose when the form-step stuff gets built out, we won't need to attach `data` here? My understanding based on our conversation was that the data / payload would get delegated down to the associated `<form-step>` to do its setup.

Reply:
Yeah I think that'll probably be the way to go. For now this made it possible to test and be sure `setStep` was doing what it needed to do. Once the `<form-step>` piece is in place I'll make the updates and change the tests to validate that `setState` gets called with `data` (or something along those lines).